milli/src/update/index_documents/extract/extract_geo_points.rs

```diff
@@ -30,39 +31,74 @@ pub fn extract_geo_points<R: io::Read + io::Seek>(
     let mut cursor = obkv_documents.into_cursor()?;
     while let Some((docid_bytes, value)) = cursor.move_on_next()? {
         let obkv = obkv::KvReader::new(value);
-        // since we only needs the primary key when we throw an error we create this getter to
-        // lazily get it when needed
+        // since we only need the primary key when we throw an error
+        // we create this getter to lazily get it when needed
         let document_id = || -> Value {
             let document_id = obkv.get(primary_key_id).unwrap();
             serde_json::from_slice(document_id).unwrap()
         };
 
-        // first we get the two fields
-        let lat = obkv.get(lat_fid);
-        let lng = obkv.get(lng_fid);
+        // HELP we will receive two DelAdds here, one for the lat and one for the lng
+        //      what happens if there is a missing Del or Add for one of them?
```

Comment:
I need your attention, here 👀

Reply:
Hello @Kerollmops,
In the previous behavior, if you are **replacing** a document with a missing latitude, Meilisearch would throw an error, I think your current version wouldn't.
let's say I have a document:
`{"id": 1, "_geo": { "lat": 0.0, "lng": 0.0 }}`
that we **replace** by:
`{"id": 1, "_geo": { "lat": 1.0 }}`
 
 The DelAdd version of the document would be:
```
"_geo.lat": {"Del": 0.0, "Add": 1.0 }, 
"_geo.lng": {"Del": 0.0 }
```
Which would be interpreted as:
1) line 45-46: `(Some({"Del": 0.0, "Add": 1.0 }), Some({"Del": 0.0 }))`
2) line 51: `del_lat_lng = Some(<extracted_lat_lng>)` because lat.Del is Some and lng.Del is Some
3) line 56: `del_lat_lng = None` because lat.Add is Some but lng.Add is None
4) line 64: insert only the deletion in the Sorter and continue

Whereas the previous version would have directly jumped to line 81-83 `return Err(GeoError::MissingLongitude { document_id: document_id() }.into())`.

However, the case where lat or lng are missing or are not well formatted is not possible because the document is checked earlier in [enrich](https://github.com/meilisearch/meilisearch/blob/cd40f6b28c4d5167b1e97152a59fcc0333c8f395/milli/src/update/index_documents/enrich.rs#L375C1-L403). Meaning that the error-throwing cases of the current `match` should be unreachable. 🤔

Reply:
I will try to make it return the right error message, thank you 😊

Reply:
Ok, after thinking more about this, it is impossible to reach this situation. I will keep the code I did and merge right away. It is forbidden to only specify the lat or LNG. Both are required.